web/src/user_topic_popover.ts

```diff
@@ -28,19 +32,28 @@ export function initialize() {
             popover_menus.popover_instances.change_visibility_policy = instance;
             popover_menus.on_show_prep(instance);
             const $elt = $(instance.reference).closest(".change_visibility_policy").expectOne();
-            const stream_id = $elt.attr("data-stream-id");
-            const topic_name = $elt.attr("data-topic-name");
+            const stream_id_str = $elt.attr("data-stream-id");
             $elt.addClass("visibility-policy-popover-visible");
+            assert(stream_id_str !== undefined);
+
+            stream_id = Number.parseInt(stream_id_str, 10);
+            topic_name = $elt.attr("data-topic-name")!;
 
             instance.setContent(
-                parse_html(render_change_visibility_policy_popover(instance.context)),
+                parse_html(
+                    render_change_visibility_policy_popover(
+                        popover_menus_data.get_change_visibility_policy_popover_content_context(
+                            stream_id,
+                            topic_name,
+                        ),
+                    ),
+                ),
```

Comment:
This should belong in the fixed commit. We are trying to fix a TypeScript bug ie.`instance` object is being mutated with a field (`context`) that it does not contain. Hence all the occurrences of `instance.context` should be replaced with the correct version in a single prep commit; fixing the problem.

Apologies if I wasn't able to get my point across in the first review.

Reply:
@LaPulgaaa I think I understand, but I just want to confirm.

**Prep-commit** (js-file), remove instance.context and replace with:  
instance.setContent(
    parse_html(
        render_change_visibility_policy_popover(
            popover_menus_data.get_change_visibility_policy_popover_content_context(
                Number.parseInt(stream_id, 10),
                topic_name,
            )
        )
    )
);

**Second commit** (ts-file):  
assert(stream_id_str !== undefined);
stream_id = Number.parseInt(stream_id_str, 10);
topic_name = $elt.attr("data-topic-name")!;

            instance.setContent(
                parse_html(
                    render_change_visibility_policy_popover(
                        popover_menus_data.get_change_visibility_policy_popover_content_context(
                            stream_id,
                            topic_name,
                        ),
                    ),
                ),
            );

Does that sound correct?

Reply:
Yes. LGTM!